tfjs-backend-webgpu/src/gather_webgpu.ts

```diff
@@ -37,16 +37,18 @@ export class GatherProgram implements WebGPUProgram {
     this.dispatchLayout = flatDispatchLayout(this.outputShape);
     this.dispatch = computeDispatch(
         this.dispatchLayout, this.outputShape, this.workGroupSize);
-    this.shaderKey = `gather`;
+    this.shaderKey = `gather_${aShape}`;
   }
 
   getUserCode(): string {
-    const sourceCoords = getSourceCoords(this.aShape, 'i32');
+    const sourceCoords = getSourceCoords(this.aShape);
     const userCode = `
       ${getMainHeaderAndGlobalIndexString()}
         if (index < uniforms.size) {
           let resRC = getCoordsFromIndex(index);
-          setOutputAtIndex(index, getA(${sourceCoords}));
+          let indexZ = i32(getIndices(resRC.x, resRC.z));
+          let inBounds = select(0.0, 1.0, indexZ >= 0 && indexZ < ${this.aShape[2]});
```

Comment:
Can you directly use `indexZ < aShape[2]` instead of `indexZ < ${this.aShape[2]}`? I think aShape is already a uniform value. In this case, you don't need to change the `shaderKey`.

Reply:
Thanks. Done.